tests/util/util.go

```diff
@@ -58,6 +64,96 @@ func MustRun(t *testing.T, command string, opts ...string) []byte {
 	return out
 }
 
+func Run(t *testing.T, command string, opts ...string) error {
+	_, err := exec.Command(command, opts...).CombinedOutput()
+	return err
+}
+
 func StringToPtr(str string) *string {
 	return &str
 }
+
+func FetchLocalImage(t *testing.T) string {
+	tmpPath := os.Getenv("TMPDIR")
+	if tmpPath == "" {
+		tmpPath = "/var/tmp"
+	}
+
+	tmpDir, err := ioutil.TempDir(tmpPath, "")
+	if err != nil {
+		t.Fatalf("failed creating temp dir: %v", err)
+	}
+
+	err = DownloadFile(tmpDir, "coreos_production_image.bin.bz2")
+	if err != nil {
+		os.RemoveAll(tmpDir)
+		t.Fatalf("failed downloading image: %v", err)
+	}
+
+	err = DownloadFile(tmpDir, "coreos_production_image.bin.bz2.sig")
+	if err != nil {
+		os.RemoveAll(tmpDir)
+		t.Fatalf("failed downloading signature: %v", err)
+	}
+
+	err = DownloadFile(tmpDir, "version.txt")
+	if err != nil {
+		os.RemoveAll(tmpDir)
+		t.Fatalf("failed downloading version: %v", err)
+	}
+
+	return tmpDir
+}
+
+func DownloadFile(tmpDir, name string) error {
+	file, err := os.Create(filepath.Join(tmpDir, name))
+	if err != nil {
+		return fmt.Errorf("failed to create file")
+	}
+	defer file.Close()
+
+	resp, err := http.Get(fmt.Sprintf("https://stable.release.core-os.net/amd64-usr/current/%s", name))
+	if err != nil {
+		return fmt.Errorf("failed to download file")
+	}
+	defer resp.Body.Close()
+
+	_, err = io.Copy(file, resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed copying file data")
+	}
+
+	return nil
+}
+
+type HTTPServer struct {
+	FileDir string
+}
+
+func (server *HTTPServer) Version(w http.ResponseWriter, r *http.Request) {
+	http.ServeFile(w, r, filepath.Join(server.FileDir, "version.txt"))
+}
+
+func (server *HTTPServer) Image(w http.ResponseWriter, r *http.Request) {
+	http.ServeFile(w, r, filepath.Join(server.FileDir, "coreos_production_image.bin.bz2"))
+}
+
+func (server *HTTPServer) Signature(w http.ResponseWriter, r *http.Request) {
+	http.ServeFile(w, r, filepath.Join(server.FileDir, "coreos_production_image.bin.bz2.sig"))
+}
+
+func (server *HTTPServer) Start(t *testing.T) {
+	http.HandleFunc("/current/version.txt", server.Version)
+
+	data, err := ioutil.ReadFile(filepath.Join(server.FileDir, "version.txt"))
+	if err != nil {
+		t.Fatalf("Couldn't read version.txt")
+	}
+	version := RegexpSearch(t, "version", "COREOS_VERSION=(.*)", data)
+
+	http.HandleFunc(fmt.Sprintf("/%s/coreos_production_image.bin.bz2", version), server.Image)
+	http.HandleFunc(fmt.Sprintf("/%s/coreos_production_image.bin.bz2.sig", version), server.Signature)
+
+	s := &http.Server{Addr: ":8080"}
```

Comment:
You'll probably want to listen on localhost only.  Also, in general, hardcoded ports should be avoided for this sort of thing, since they prevent two copies of the test suite from running at once.  You can [have the OS pick an unused port](https://stackoverflow.com/a/43425461) and pass it through to the tests.

(I just realized the Ignition tests also hardcode the server port...)